cmd/oras/root/attach.go

```diff
@@ -164,7 +167,14 @@ func runAttach(ctx context.Context, opts attachOptions) error {
 
 	root, err := pushArtifact(dst, pack, copy)
 	if err != nil {
-		return err
+		var re *remote.ReferrersError
+		if !errors.As(err, &re) || !re.IsReferrersIndexDelete() {
+			// not referrer index delete error
+			if opts.strict {
+				return err
+			}
+		}
+		logger.Info("Attach is successful but removal of outdated referrers index from remote registry failed. Garbage collection may be required.")
```

Comment:
I would suggest revising it to **Attached successfully but the removal of outdated referrers index from the remote registry failed. Garbage collection may be required.**
since [attach is always used as a verb](https://www.dictionary.com/browse/attach).

Reply:
Good catch, resolved